src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -355,22 +396,24 @@ private void HandleSubscription(KafkaConsumerActorMetadata.Internal.ISubscriptio
                         CheckOverlappingRequests("Assign", Sender, assign.TopicPartitions);
                         var previousAssigned = _consumer.Assignment;
                         _consumer.Assign(assign.TopicPartitions.Union(previousAssigned));
-                        _commitRefreshing.AssignedPositions(assign.TopicPartitions, _consumer, _settings.PositionTimeout);
+                        _commitRefreshing.AssignedPositions(assign.TopicPartitions, _consumer,
+                            _settings.PositionTimeout);
                         break;
                     }
 
                     case KafkaConsumerActorMetadata.Internal.AssignWithOffset assignWithOffset:
                     {
-                        var topicPartitions = assignWithOffset.TopicPartitionOffsets.Select(o => o.TopicPartition).ToImmutableHashSet();
+                        var topicPartitions = assignWithOffset.TopicPartitionOffsets.Select(o => o.TopicPartition)
+                            .ToImmutableHashSet();
                         CheckOverlappingRequests("AssignWithOffset", Sender, topicPartitions);
-                        
-                        // TODO: dear lord this is wrong, WE SHOULD NOT BE SETTING OFFSETS TO ZERO HERE
-                        var previousAssigned = _consumer.Assignment.Select(tp => new TopicPartitionOffset(tp, new Offset(0)));
```

Comment:
Fixed this code - explicitly assigning `0` here is a no-no. Instead we use whatever the most recently stored offset for this consumer on this partition was. If the consumer is using external offset storage then this should default to `Offset.Unassigned`.